cfgen/cfgen

```diff
@@ -1956,8 +1948,66 @@ def build_cluster(cluster_desc: Dict[str, Any]) -> Cluster:
                 proc_id = ConfProcess.build(conf, node_id, node, proc_t)
                 cluster.m0_clients[proc_id] = new_oid(ObjT.service)
 
+    # Create data structures required for aux pools
+    aux_pools: dict = {}
+    disk_ref_combinations = []
+    encl_ref_combination = []
+    tolerance: Failures
+    pool_nodes: list = []
+
     for pool in cluster_desc['pools']:
-        ConfPool.build(conf, root_id, pool)
+        pool_t = pool_type(pool)
+        if pool_t == PoolT.sns:
+            # Calculate and create aux pool disk combinations
+            # only for SNS pool type
+            d = pool.get('allowed_failures')
+            # Calculate the tolerance in case of SNS pools
+            if d:
+                tolerance = Failures(d['site'], d['rack'], d['encl'],
+                                     d['ctrl'], d['disk'])
+            else:
+                tolerance = ConfPool.gen_sns_tolerance(conf, pool)
+            # Generate the node combinations considering node failures as
+            # allowed by node tolerance values
+            temp_pool_nodes = []
+            for nodes in pool['disk_refs']:
+                temp_pool_nodes += [
+                    nodes['node']]
+            pool_nodes = list(set(temp_pool_nodes))
```

Comment:
Is it just a way to guarantee the uniqueness in poll_nodes?

Reply:
I guess not. There are multiple ways to find unique ones.
But I felt this is simple technique to get only unique ones in the list.